CoffeeChat/Controller/DemographicsViewController.swift

```diff
@@ -8,23 +8,36 @@
 
 import UIKit
 
+enum DropdownView {
+    case search, select
+}
+
 class DemographicsViewController: UIViewController {
 
     // MARK: - Private Data vars
-    private weak var delegate: OnboardingPageDelegate?
+    private var activeDropdownViewType: DropdownView?
+    private var classSearchFields: [String] = []
+    private weak var delegate: OnboardingPageDelegate!
+    private var fieldsEntered: [Bool] = [false, false, false, false] // Keep track of selection status of each field.
+    // TODO: Update with networking values from backend
+    private let hometownSearchFields = ["Boston, MA", "New York, NY", "Washington, DC", "Sacramento, CA", "Ithaca, NY"]
+    private let majorSearchFields = ["Computer Science", "Economics", "Psychology", "English", "Government"]
+    private let pronounSearchFields = ["She/Her/Hers", "He/Him/His", "They/Them/Theirs"]
     private let userDefaults = UserDefaults.standard
 
     // MARK: - Private View Vars
-    private let classTextField = UITextField()
+    private var activeDropdownView: UIView?
+    private var classDropdownView: OnboardingSelectDropdownView!
     private let greetingLabel = UILabel()
     private let helloLabel = UILabel()
-    private let hometownSearchBar = UISearchBar()
-    private let majorSearchBar = UISearchBar()
+    private var hometownDropdownView: OnboardingSearchDropdownView!
```

Comment:
is it possible to avoid the force unwrap?

Reply:
Hm, not sure. To initialize the dropdownview, I have to pass in a delegate but I can't use self until the view controller has been initialized so there's some strange loop logic I'm not sure I can get around?